posthog/api/capture.py

```diff
@@ -196,6 +196,8 @@ def _get_sent_at(data, request) -> Tuple[Optional[datetime], Any]:
 def _check_token_shape(token: Optional[str]) -> Optional[str]:
```

Comment:
```suggestion
def _check_token_shape(token: Any) -> Optional[str]:
```

Reply:
`Any` would allow any type, thus not raise any errors regardless of what is passed. I think the current one is correct?

Reply:
[Linting did fail](https://github.com/PostHog/posthog/actions/runs/4294572841/jobs/7483824657) with `Optional[str]`, because the "is not null, is not string" is dead code to the linter. Moving to `Any` does fix this.

Reply:
Why would this not be `None` or `str`? Seems like we are calling the function from the return value of `get_token` which is typed as `Optional[str]`.

Reply:
What I'm saying is that if `get_token` returns `Optional[str]`, then it is indeed dead code. If it doesn't return `Optional[str]` then the type hint should be modified to reflect that.

Reply:
`get_token` may say that it's going to return a string, but there is no runtime validation of that fact. It can actually be any JSON value.

Reply:
Exactly, then it should be type hinted as `Any` too (or a JSON value).

Reply:
Is there something like an `unknown` type hint that would require runtime type checking?

Reply:
I would say no type hint then :sweat_smile: